### DISCUSSION TITLE
Why does -I cause "Item max size cannot be less than 1024 bytes." errors?

### DISCUSSION THREAD

**Discussion by jerrac** [memcached/memcached#1066] [Q&A] [answered]
I'm brand new to Memcached. Google hasn't found any answers, so I figured this might be a good place to ask.

When I try to run Memcached with the `-I/--max-item-size` flag, it immediate throws out that message and then exits.

```
docker run -it --rm memcached:1.6.21 bash 
memcache@a3da14171df6:/$ memcached -I 2147483648
Item max size cannot be less than 1024 bytes.
memcache@a3da14171df6:/$ 
```
2048m and 2048M produce the same results.

The help output says:

```
-I, --max-item-size=<num> adjusts max item size
                          (default: 1m, min: 1k, max: 1024m)
```
Which I assume means that the default max size is 1G. 

I'm specifically trying to set up a better cache backend for a Drupal 9 site that has cache tables in excess of 1G. So I figured letting it use up to 2G would cover my needs.

Anyway, why is that error popping up?

**Comment by dormando** [comment]
`-I` changes the maximum size of a single cache item entry. I'm guessing you actually want the `-m` option. set that to `-m 2048`

**Reply by jerrac** [reply]
Thanks for the reply, but I am actually aiming to increase the size a **single** entry can be. I did use `-m` to increase how much total memory memcached can use. I just left that out of my post to keep things focused.

**Reply by dormando** [reply]
well for one it says in the help output you quoted that the max size is 1024m and you're trying to double that.

I know you're new to the system so please take this hint; there's absolutely no way you need to increase the single item size limit. You're misunderstanding something with your drupal configuration. There's no way a system would function if it had to pull an entire two gigabytes over a socket every time it was accessing something. Page loadtime would be minutes,

Tables are split into multiple smaller items related to the sub-data they represent.

Unless you're getting a very specific error from memcached about the items you're storing being too large, you don't need to adjust `-I`

**Reply by jerrac** [reply]
So the max is 1gb period, not that the default max size is 1gb? The way I read the quoted docs is that those numbers are the defaults. Not that they are the actual limits.

Neither the docs, nor the error message, indicate that the issue is that I'm trying to set the number higher than it can go. The error message actually indicates I'm making it too small... 

As for my specific Drupal issue, I think you're right that none of the items are over 1gb, since I went ahead and set it up leaving `-I` off my startup command. But I am dealing with a stupidly huge menu cache entry that is causing my mariadb bin logs to eat up 20-30gb in less than 3 hours. (I can't make the cache expiration as long as I'd like... ) The cache tables have reached 2+gb in size before. Which is where the thought that an item might get that large came from.

**Reply by dormando** [reply, marked as answer]
you're getting that error because you've set a value so high it's overflowing to zero. That is a bug I should fix.

The default limit for an individual item is 1m. again, that help output you quoted shows default of 1 megabyte, min 1k, max 1gb.

This is a forum about memcached as a daemon, so I'm not an expert in drupal or how drupal uses cache. Have you asked a drupal community for help? I don't know what a "cache table" is or how that relates to the size of a particular item.

**Reply by dormando** [reply]
I wonder if you're misreading the help output as the size range of an item?

`-I` relates to the _maximum size of a particular item_ it doesn't have anything to do with the range or the minimum. By default you can store anything from a few bytes to 1 megabyte per item, which is usually plenty.

**Comment by aa-shalin** [comment]
For me, it had to do with the order in which args were passed. 
`-I 32m -m 8192` resulted in `Item max size cannot be less than 1024 bytes.` 
But `-m 8192 -I 32m` worked fine.
